### PR TITLE
test/data_manager: concurrent mutations

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

File: tests/data_manager.rs [hunk 1]
```diff
@@ -689,6 +690,280 @@ fn mutable_data_parallel_mutations() {
...
+    verify_data_is_stored(&mut nodes, &mut clients[0], &all_data);
+}
+
+// Two clients concurrently mutating same data with same actions, one with permission, the other not.
+// Only the one with permission shall succeed.
+#[test]
+fn no_permission_mutable_data_concurrent_mutations() {
+    let seed = None;
+    let node_count = TEST_NET_SIZE;
+    let client_count = 2;
```
Comment (madadam) [inline tests/data_manager.rs]: Minor: I don't think we need to extract the client count here, as this test requires precisely two clients. I'd say it's fine to hardcode it on line 849.

File: tests/data_manager.rs [hunk 2]
```diff
@@ -689,6 +690,280 @@ fn mutable_data_parallel_mutations() {
...
+
+    for client in &mut clients {
+        client.ensure_connected(&mut nodes);
+        client.create_account(&mut nodes);
+    }
+
+    // Put some data to the network.
+    let mut all_data = vec![];
+
+    // Allow any mutations for anyone.
```
Comment (madadam) [inline tests/data_manager.rs]: It should be "Allow mutations only for the first client" or something like that.

File: tests/data_manager.rs [hunk 3]
```diff
@@ -689,6 +690,280 @@ fn mutable_data_parallel_mutations() {
...
+                       response: Response::MutateMDataEntries { res, .. }, ..
+                   } = event {
+                network_responded = true;
+                match res {
+                    Ok(()) => {
+                        panic!("Client {:?} shall not receive successful response.",
+                               clients[1].name());
+                    }
+                    Err(error) => {
+                        trace!("Client {:?} received failed response. Reason: {:?}",
```
Comment (madadam) [inline tests/data_manager.rs]: I'd assert the expected error here (I think `ClientError::AccessDenied`).

File: tests/data_manager.rs [hunk 4]
```diff
@@ -689,6 +690,280 @@ fn mutable_data_parallel_mutations() {
...
 
+// Client trying to mutate a mutable_data concurrently with two different sets of actions.
+// The responses to the two mutate attempts shall be:
+//  1, both succeeded, when there is no conflicting mutation
+//  2, only one succeeded, when there is conflicting mutation (updating the same key)
+#[test]
+fn mutable_data_concurrent_mutations() {
+    let seed = None;
+    let node_count = TEST_NET_SIZE;
+    let client_count = 1;
```
Comment (madadam) [inline tests/data_manager.rs]: This test uses only one client, so this line could be removed.

File: tests/data_manager.rs [hunk 5]
```diff
@@ -689,6 +690,280 @@ fn mutable_data_parallel_mutations() {
...
+    let data_count = 5;
+    let iterations = test_utils::iterations();
+
+    let network = Network::new(GROUP_SIZE, seed);
+    let mut rng = network.new_rng();
+    let mut event_count = 0;
+    let mut nodes = test_node::create_nodes(&network, node_count, None, false);
+    let mut clients: Vec<_> = (0..client_count)
```
Comment (madadam) [inline tests/data_manager.rs]: Again, only one client used. This can be simplified to just `let client = TestClient::new(...)`.

**Review by madadam — CHANGES_REQUESTED** [review]

File: tests/data_manager.rs (hunk 2 above)
Comment (maqi) [reply tests/data_manager.rs]: i think the comment here is correct

Comment (madadam) [reply tests/data_manager.rs]: Yes. It was wrong in the other test, I probably commented here by mistake. All good now.

**Review by madadam — APPROVED** [review]

**Pull request merged**